core/src/main/scala/org/apache/spark/eventhubs/client/EventHubsClient.scala

```diff
@@ -238,20 +242,31 @@ private[spark] class EventHubsClient(private val ehConf: EventHubsConf)
           case StartOfStream => (nAndP.partitionId, earliestSeqNoF(nAndP.partitionId))
           case EndOfStream   => (nAndP.partitionId, latestSeqNoF(nAndP.partitionId))
           case _ =>
-            val receiver = retryJava(client.createEpochReceiver(consumerGroup,
-                                                                nAndP.partitionId.toString,
-                                                                pos.convert,
-                                                                DefaultEpoch),
-                                     "translate: epoch receiver creation.")
-            val seqNo = receiver
-              .flatMap { r =>
-                retryNotNull(r.receive(1), "translate: receive call")
-              }
-              .map { e =>
-                e.iterator.next.getSystemProperties.getSequenceNumber
+            val runtimeInfo =
+              Await.result(getRunTimeInfoF(nAndP.partitionId), ehConf.internalOperationTimeout)
+            val seqNo =
+              if (runtimeInfo.getIsEmpty || (pos.enqueuedTime != null &&
+                  runtimeInfo.getLastEnqueuedTimeUtc.isBefore(pos.enqueuedTime.toInstant))) {
+                latestSeqNoF(nAndP.partitionId)
```

Comment:
Is `latestSeqNo` already available in `runtimeInfo`?

Reply:
good catch, yes, we can get it from the runtimeInfo.